src/pages/RFOX/hooks/useGetUnstakingRequestQuery.ts

```diff
@@ -66,21 +72,29 @@ export const useGetUnstakingRequestQuery = <SelectData = UnstakingRequest>({
     [contracts],
   )
 
-  const getUnstakingRequestQueryFn = useMemo(
-    () =>
-      unstakingRequestCountResponse && unstakingRequestCountResponse?.valueOf() > 0n
-        ? () =>
-            multicall(client, {
-              contracts,
-            }).then(r => r.map(response => response.result).filter(isSome))
-        : skipToken,
-    [contracts, unstakingRequestCountResponse],
-  )
+  const getUnstakingRequestQueryFn = useMemo(() => {
+    if (isUnstakingRequestCountLoading || isUnstakingRequestCountPending) return skipToken
```

Comment:
We're *actually* loading/pending, fine to not have a query here and `skipToken` - this query will be in `pending` state, which is correct.

Reply:
It feels a bit complex to understand on my side as I think we shouldn't even run this query if required dependencies aren't up yet

I don't know if it's clear but It feel super hacky to link the states of the linked dependency with the actual state of this query

The end code is elegant but it's not so easy to understand from an external point of view, I would expect to early return the state of `useGetUnstakingRequestCountQuery` instead of running the unstakingRequest query in case it's loading or pending but at the same time it could even be more complex considering the fact that we might expect a query to be returned anyway

Happy with the actual implementation but can we add some comments to say that we are pretty linking the states of both queries together?

Reply:
Will propagate the comments from this PR @NeOMakinG!

Reply:
45469b628aaa7da5875a2334dce696cf775b19a7